tests/app/celery/test_service_callback_tasks.py

```diff
@@ -7,35 +7,141 @@
+from app import (DATETIME_FORMAT, encryption)
 
-from tests.app.conftest import (
-    sample_service as create_sample_service,
-    sample_template as create_sample_template,
-)
 from tests.app.db import (
     create_notification,
-    create_user,
-    create_service_callback_api
+    create_service_callback_api,
+    create_service,
+    create_template
 )
 from app.celery.service_callback_tasks import send_delivery_status_to_service
 from app.config import QueueNames
 
 
 @pytest.mark.parametrize("notification_type",
                          ["email", "letter", "sms"])
-def test_send_delivery_status_to_service_post_https_request_to_service(notify_db,
-                                                                       notify_db_session,
-                                                                       notification_type):
-    user = create_user()
-    service = create_sample_service(notify_db, notify_db_session, user=user, restricted=True)
+def test_send_delivery_status_to_service_post_https_request_to_service_with_encrypted_data(
+        notify_db_session, notification_type):
+
+    callback_api, template = _set_up_test_data(notification_type)
+    datestr = datetime(2017, 6, 20)
+
+    notification = create_notification(template=template,
+                                       created_at=datestr,
+                                       updated_at=datestr,
+                                       sent_at=datestr,
+                                       status='sent'
+                                       )
+    encrypted_status_update = _set_up_encrypted_data(callback_api, notification)
+    with requests_mock.Mocker() as request_mock:
+        request_mock.post(callback_api.url,
+                          json={},
+                          status_code=200)
+        send_delivery_status_to_service(notification.id, encrypted_status_update=encrypted_status_update)
 
+    mock_data = {
+        "id": str(notification.id),
+        "reference": notification.client_reference,
+        "to": notification.to,
+        "status": notification.status,
+        "created_at": datestr.strftime(DATETIME_FORMAT),
+        "completed_at": datestr.strftime(DATETIME_FORMAT),
+        "sent_at": datestr.strftime(DATETIME_FORMAT),
+        "notification_type": notification_type
+    }
+
+    assert request_mock.call_count == 1
+    assert request_mock.request_history[0].url == callback_api.url
+    assert request_mock.request_history[0].method == 'POST'
+    assert request_mock.request_history[0].text == json.dumps(mock_data)
+    assert request_mock.request_history[0].headers["Content-type"] == "application/json"
+    assert request_mock.request_history[0].headers["Authorization"] == "Bearer {}".format(callback_api.bearer_token)
+
+
+@pytest.mark.parametrize("notification_type",
+                         ["email", "letter", "sms"])
+def test_send_delivery_status_to_service_retries_if_request_returns_500_with_encrypted_data(
```

Comment:
should probs add a new test to make sure it retries if the DB errors too.

Reply:
There are no hits to the db

Reply:
duhhhh 👍